src/optimizer/plan_generation.c

```diff
@@ -1352,6 +1356,24 @@ make_pred_from_plan (QO_ENV * env, QO_PLAN * plan, PT_NODE ** key_predp, PT_NODE
   /* make predicate list for key filter */
   if (key_predp != NULL)
     {
+      if (qo_index_infop->need_copy_multi_range_term != -1 && qo_index_infop->need_copy_to_sarg_term)
+	{
+	  bitset_add (&(plan->sarged_terms), qo_index_infop->need_copy_multi_range_term);
+	}
+      else if (qo_index_infop->need_copy_multi_range_term != -1)
+	{
+	  index_entryp = qo_index_infop->ni_entry->head;
+	  if (index_entryp && index_entryp->constraints->func_index_info && index_entryp->cover_segments == false)
```

Comment:
might need to check if index_entryp->constraints is null ?

Reply:
fixed